test_runner/src/main/kotlin/ftl/run/platform/android/GetInstrumentationShardChunks.kt

```diff
@@ -0,0 +1,54 @@
+package ftl.run.platform.android
+
+import com.linkedin.dex.parser.DexParser
+import ftl.args.AndroidArgs
+import ftl.args.ArgsHelper
+import ftl.args.ShardChunks
+import ftl.config.FtlConstants
+import ftl.filter.TestFilter
+import ftl.filter.TestFilters
+import ftl.run.model.InstrumentationTestApk
+import ftl.util.FlankTestMethod
+import ftl.util.downloadIfNeeded
+import java.io.File
+
+fun getInstrumentationShardChunks(
+    args: AndroidArgs,
+    testApks: List<InstrumentationTestApk>
+): Map<InstrumentationTestApk, ShardChunks> =
+    getFlankTestMethods(
+        testApks = testApks.download(),
+        testFilter = TestFilters.fromTestTargets(args.testTargets)
+    ).mapValues { (_, testMethods: List<FlankTestMethod>) ->
+        when {
+            testMethods.isEmpty() -> emptyList<List<String>>().also { printNoTests(testApks) }
+            args.numUniformShards == null -> ArgsHelper.calculateShards(testMethods, args)
+            else -> listOf(testMethods.map(FlankTestMethod::testName))
+        }
+    }
+
+private fun getFlankTestMethods(
+    testApks: List<InstrumentationTestApk>,
+    testFilter: TestFilter
+): Map<InstrumentationTestApk, List<FlankTestMethod>> =
+    testApks.associateWith { testApk ->
+        DexParser.findTestMethods(testApk.test.local).asSequence().distinct().filter(testFilter.shouldRun).map { testMethod ->
```

Comment:
How about more functional styling here, WDYT?

Reply:
There are only function calls, no value assignments, no ordinary loops, even no conditions so I have no idea how to do it more functional :). Pls explain me by example.

Reply:
Well my comment was just cosmetic one. Here we have invocation chain in one line, I was thinking about splitting it to new line each. So as I said 100% cosmetic